pkg/pipeline/service.go

```diff
@@ -363,3 +366,14 @@ func (s *Service) validatePipeline(cfg Config, id string) error {
 
 	return cerrors.Join(errs...)
 }
+
+func (s *Service) updateOldStatusMetrics(pl *Instance) {
+	status := strings.ToLower(pl.GetStatus().String())
+	measure.PipelinesGauge.WithValues(status).Dec()
```

Comment:
Wait, shouldn't `WithValues` get `pl.Name`? And the gauge then needs to be set to the correct status value.

Reply:
Not really, because this is actually this line: https://github.com/ConduitIO/conduit/pull/1828/files#diff-4d290fa258ba1313fd6cb7e6393c2efa98c5fae36e7e0d7ace77f31bc6c53557L596

It's updating the metrics for the old status value, before the new value is set and then the metrics for the new value are updated.

Reply:
Oh I see, that's the existing pipelines gauge, not the pipeline status gauge, my bad!